File: Java/Program/Kosan.java
import java.util.ArrayList;

//Class Kosan simpan data kosan
public class Kosan{
    private int jumlahKamar;
    private String kodeKosan;
    private String namaKosan;
    private String alamatKosan;
    private ArrayList<Kamar> listKamar = new ArrayList<>();
    private ArrayList<KamarPremium> listKamarPremium = new ArrayList<>();
    private ArrayList<FasilitasUmum> listFasilitasUmum = new ArrayList<>();

    //Constructor kosong
    public Kosan(){
        this.kodeKosan = "";
        this.namaKosan = "";
        this.alamatKosan = "";
        this.jumlahKamar = 0;
    }

    //Constructor lengkap
    public Kosan(String kodeKosan, String namaKosan, String alamatKosan, int jumlahKamar){
        this.kodeKosan = kodeKosan;
        this.namaKosan = namaKosan;
        this.alamatKosan = alamatKosan;
        this.jumlahKamar = jumlahKamar;
    }

    //Getter dan Setter untuk setiap atribut
    public String getKodeKosan(){
        return kodeKosan;
    }
    public void setKodeKosan(String kodeKosan){
        this.kodeKosan = kodeKosan;
    }

    public String getNamaKosan(){
        return namaKosan;
    }
    public void setNamaKosan(String namaKosan){
        this.namaKosan = namaKosan;
    }

    public String getAlamatKosan(){
        return alamatKosan;
    }
    public void setAlamatKosan(String alamatKosan){
        this.alamatKosan = alamatKosan;
    }

    public int getJumlahKamar(){
        return jumlahKamar;
    }
    public void setJumlahKamar(int jumlahKamar){
        this.jumlahKamar = jumlahKamar;
    }

    //Tambah Kamar & Fasilitas
    public ArrayList<Kamar> getListKamar(){
        return listKamar;
    }
    public void tambahKamar(Kamar k){
        listKamar.add(k);
    }

    public ArrayList<KamarPremium> getListKamarPremium(){
        return listKamarPremium;
    }
    public void tambahKamarPremium(KamarPremium kp){
        listKamarPremium.add(kp);
    }

    public ArrayList<FasilitasUmum> getListFasilitasUmum(){
        return listFasilitasUmum;
    }
    public void tambahFasilitasUmum(FasilitasUmum f){
        listFasilitasUmum.add(f);
    }
}

File: CPP/Program/Kosan.cpp
#pragma once
#include <vector>
#include "Kamar.cpp"
#include "KamarPremium.cpp"
#include "FasilitasUmum.cpp"
using namespace std;

//Class Kosan untuk simpan data kosan, daftar kamar dan fasilitas
class Kosan{

    private:
    int jumlahKamar;
    string kodeKosan, namaKosan, alamatKosan;
    vector<Kamar> listKamar;
    vector<KamarPremium> listKamarPremium;
    vector<FasilitasUmum> listFasilitasUmum;

    public:
    //Constructor kosong
    Kosan(){
        this->kodeKosan = "";
        this->namaKosan = "";
        this->alamatKosan = "";
        this->jumlahKamar = 0;
    }

    //Constructor dengan parameter untuk buat Kosan dengan data
    Kosan(string kodeKosan, string namaKosan, string alamatKosan, int jumlahKamar){
        this->kodeKosan = kodeKosan;
        this->namaKosan = namaKosan;
        this->alamatKosan = alamatKosan;
        this->jumlahKamar = jumlahKamar;
    }

    string getKodeKosan(){
        return kodeKosan;
    }
    void setKodeKosan(string kodeKosan){
        this->kodeKosan = kodeKosan;
    }

    string getNamaKosan(){
        return namaKosan;
    }
    void setNamaKosan(string namaKosan){
        this->namaKosan = namaKosan;
    }

    string getAlamatKosan(){
        return alamatKosan;
    }
    void setAlamatKosan(string alamatKosan){
        this->alamatKosan = alamatKosan;
    }

    int getJumlahKamar(){
        return jumlahKamar;
    }
    void setJumlahKamar(int jumlahKamar){
        this->jumlahKamar = jumlahKamar;
    }

    //Method untuk tambah data ke list
    vector<Kamar> getListKamar(){
        return listKamar;
    }
    void TambahKamar(Kamar k){
        listKamar.push_back(k);
    }

    vector<KamarPremium> getListKamarPremium(){
        return listKamarPremium;
    }
    void TambahKamarPremium(KamarPremium kp){
        listKamarPremium.push_back(kp);
    }

    vector<FasilitasUmum> getListFasilitasUmum(){
        return listFasilitasUmum;
    }
    void TambahFasilitasUmum(FasilitasUmum f){
        listFasilitasUmum.push_back(f);
    }

    //Destruktor
    ~Kosan(){}
};